src/Microsoft.Health.Operations.UnitTests/Serialization/UtcCompatibilityJsonConverterTests.cs

```diff
@@ -0,0 +1,86 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using Microsoft.Health.Operations.Serialization;
+using Xunit;
+
+namespace Microsoft.Health.Operations.UnitTests.Serialization;
+
+public class UtcCompatibilityJsonConverterTests
+{
+    private static readonly JsonSerializerOptions DefaultOptions = new();
+
+    [Theory]
+    [InlineData("null")]
+    [InlineData("42")]
+    [InlineData("{ \"foo\": \"bar\" }")]
+    [InlineData("[ 1, 2, 3 ]")]
+    [InlineData("\"\"")]
+    [InlineData("\"bar\"")]
+    public void GivenInvalidToken_WhenReadingJson_ThenThrowJsonReaderException(string json)
+    {
+        Assert.Throws<JsonException>(() =>
+        {
+            Utf8JsonReader jsonReader = new(Encoding.UTF8.GetBytes(json));
+            Assert.True(jsonReader.Read());
+            new UtcCompatibilityJsonConverter().Read(ref jsonReader, typeof(DateTimeOffset), DefaultOptions);
+        });
+    }
+
+    [Theory]
+    [InlineData("\"1234-05-06T07:08:09Z\"")]
+    [InlineData("\"1234-05-06T07:08:09+00:00\"")]
+    public void GivenValidToken_WhenReadingJson_ThenReturnDateTimeOffset(string json)
+    {
+        DateTimeOffset expected = new(1234, 5, 6, 7, 8, 9, TimeSpan.Zero);
+        Utf8JsonReader jsonReader = new(Encoding.UTF8.GetBytes(json));
+
+        Assert.True(jsonReader.Read());
+        DateTimeOffset actual = new UtcCompatibilityJsonConverter().Read(ref jsonReader, typeof(DateTimeOffset), DefaultOptions);
+        Assert.Equal(expected, actual);
+    }
+
+    [Fact]
+    public void GivenUtcDateTimeOffset_WhenWritingJson_ThenWriteZ()
```

Comment:
nit: Should this be something like `_WriteUtc` or `_WriteUtcDateTimeOffset`?

Reply:
No, we explicitly want it to be a `Z` instead of the offset